development/prow-installer/pkg/kubectl/wrapper.go

```diff
@@ -0,0 +1,104 @@
+package kubectl
+
+import (
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+type Wrapper struct {
+	Kubeconfig string
+}
+
+func (c *Wrapper) Apply() *KubeApplyCommand {
+	return &KubeApplyCommand{Config: c.Kubeconfig}
+}
+
+type KubeApplyCommand struct {
+	Type   string
+	Path   string
+	Nspace string
+	Config string
+}
+
+func (kac *KubeApplyCommand) File(path string) *KubeApplyCommand {
+	kac.Type = "-f" //file
+	//change slashed path (dir/filename) to OS-dependent path.
+	kac.Path = filepath.FromSlash(path)
+	return kac
+}
+
+func (kac *KubeApplyCommand) Directory(path string) *KubeApplyCommand {
+	kac.Type = "-k" //directory
+	//change slashed path (dir/directory/) to OS-dependent path.
+	kac.Path = filepath.FromSlash(path)
+	return kac
+}
+
+func (kac *KubeApplyCommand) Do() error {
+	cmd := exec.Command("kubectl", "apply")
+	cmd.Env = append(os.Environ(), fmt.Sprintf("KUBECONFIG=%s", kac.Config))
+
+	if kac.Path == "" {
+		return fmt.Errorf("path cannot be empty")
+	}
+
+	cmd.Args = append(cmd.Args, kac.Type, kac.Path)
+
+	log.Debugf("executing command: %s", cmd.String())
+	if out, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("error executing command: %w %s", err, out)
+	}
+	return nil
+}
+
+// generate kubeconfig based on credentials provided in arguments
+// the function returns path to the config file.
+// it's needed to have GOOGLE_CREDENTIALS_APPLICATION env variable set
+func GenerateKubeconfig(endpoint, cadata, name string) (string, error) {
+	if endpoint == "" {
+		return "", fmt.Errorf("endpoint cannot be empty")
+	}
+	if cadata == "" {
+		return "", fmt.Errorf("cadata cannot be empty")
+	}
+	if name == "" {
+		return "", fmt.Errorf("name cannot be empty")
+	}
+
+	path := filepath.FromSlash(".kube/" + name + "_config")
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		if err := os.MkdirAll(".kube", 0700); err != nil {
+			return "", fmt.Errorf("unexpected error during folder creation %w", err)
+		}
+	}
+	f, err := os.Create(path)
+	if err != nil {
+		return "", fmt.Errorf("error creating kubeconfig file %w", err)
+	}
+	defer f.Close()
+	kubeconfigTemplate := fmt.Sprintf(`apiVersion: v1
```

Comment:
Can we have this as a template file outside of the sources, so that we can change it without touching go code?

Reply:
As for now it should be easier to leave it as it is and think about it later I guess.